opensrp-eusm/src/main/java/org/smartregister/eusm/fragment/EusmDownloadedOfflineMapsFragment.java

```diff
@@ -0,0 +1,172 @@
+        for (OfflineMapModel offlineMapModel : downloadedOfflineMapModelList) {
+            if (offlineMapModel.getOfflineMapStatus().equals(OfflineMapModel.OfflineMapStatus.SELECTED_FOR_DOWNLOAD)) {
+                mapsToDownload.add(offlineMapModel);
+            }
+        }
+
+        if (mapsToDownload.isEmpty()) {
+            displaySnackBar(getString(R.string.select_offline_map_to_delete));
+            return;
+        }
+        fragmentPresenter.onDeleteDownloadMap(mapsToDownload);
+    }
+
+    private void initAdapter(RecyclerView downloadedMapsRecyclerView) {
+        adapter = new EUSMDownloadedOfflineMapAdapter(this.getContext(), this);
+        downloadedMapsRecyclerView.setAdapter(adapter);
+        if (downloadedOfflineMapModelList != null) {
+            setDownloadedOfflineMapModelList(downloadedOfflineMapModelList);
+        }
+    }
+
+    @Override
+    public void updateOfflineMapsTodelete(View view) {
+        CheckBox checkBox = (CheckBox) view;
+        OfflineMapModel offlineMapModel = (OfflineMapModel) view.getTag(R.id.offline_map_checkbox);
+
+        for (OfflineMapModel model : downloadedOfflineMapModelList) {
+            if (model.getDownloadAreaId().equals(offlineMapModel.getDownloadAreaId())) {
+                if (checkBox.isChecked()) {
+                    model.setOfflineMapStatus(OfflineMapModel.OfflineMapStatus.SELECTED_FOR_DOWNLOAD);
+                } else {
+                    model.setOfflineMapStatus(OfflineMapModel.OfflineMapStatus.READY);
+                }
+            }
+        }
+
+        setDownloadedOfflineMapModelList(downloadedOfflineMapModelList);
+    }
+
+    @Override
+    public void onDestroy() {
+        super.onDestroy();
+        fragmentPresenter = null;
+    }
+
+    @Override
+    public void deleteDownloadedOfflineMaps() {
+        displaySnackBar(getString(R.string.deleting_map));
+
+        for (OfflineMapModel offlineMapModel : downloadedOfflineMapModelList) {
+            if (offlineMapModel.getOfflineMapStatus().equals(OfflineMapModel.OfflineMapStatus.SELECTED_FOR_DOWNLOAD)) {
+                offlineMapModel.setOfflineMapStatus(OfflineMapModel.OfflineMapStatus.DOWNLOAD_STARTED);
+                OfflineServiceHelper.deleteOfflineMap(requireActivity(),
+                        offlineMapModel.getDownloadAreaId(),
+                        TaskingLibrary.getInstance().getMapboxAccessToken());
+            }
+        }
+        setDownloadedOfflineMapModelList(downloadedOfflineMapModelList);
+    }
+
+    @Override
+    public void setOfflineMapDownloadCallback(OfflineMapDownloadCallback callBack) {
+        this.callback = callBack;
+    }
+
+    @Override
+    protected void mapDeletedSuccessfully(String deletedMapName) {
+        List<OfflineMapModel> toRemove = new ArrayList<>();
+        for (OfflineMapModel offlineMapModel : downloadedOfflineMapModelList) {
+            if (offlineMapModel.getDownloadAreaId().equals(deletedMapName)) {
```

Comment:
Check for null on `deletedMapName`. Switch the args of the equal function. `(deletedMapName.equals(offlineMapModel.getDownloadAreaId))`